scripts/clang-tidy-diff.sh

```diff
@@ -4,17 +4,11 @@ set -o errexit
 
 # When we are running on travis and *not* part of a pull request we don't
 # have any upstream branch to compare against.
-if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then
-  echo "Skipping since not running on travis PR"
+if [ -z "$GITHUB_BASE_REF" ]; then
+  echo "Skipping since not running on GitHub Actions"
   exit 0
 fi
 
-if [ -n "$TRAVIS_BRANCH" ]; then
-  BRANCH=$TRAVIS_BRANCH
-else
-  BRANCH=origin/main
-fi
```

Comment:
I'd love to be able to run this script locally too.    And we have the head and base default to `HEAD` and `{upstream}` ?   I'm not sure if github actions is setup such that these will also just work there too, but maybe?

Reply:
Ideally I think the diff should look something like this `git diff "@{upstream}..."`     (using the magic upstream thing and the triple dots)

Reply:
On travis this didn't work IIRC because the upstream wasn't set correctly... but maybe it is on github?   Either way that is what we want the default / local version to be I believe.